lib/league_stat.rb

```diff
@@ -51,22 +53,48 @@ def count_of_teams
     stats_by_team.keys.size
   end
 
-  def best_offense
-    goals_per_game = {}
+  def create_scoring_averages
     stats_by_team.each do |team|
+      total_games = team[1][:total_games]
       total_goals = (team[1][:home_goals] + team[1][:away_goals]).to_f
-      goals_per_game[team[1][:team_name]] = (total_goals / team[1][:total_games]).round(2)
+      total_goals_allowed = (team[1][:home_goals_allowed] + team[1][:away_goals_allowed]).to_f
+      team[1][:total_scoring_avg] = (total_goals / total_games).round(2)
+      team[1][:total_goals_allowed_avg] = (total_goals_allowed / total_games).round(2)
+      team[1][:away_scoring_avg] = (team[1][:away_goals] / team[1][:away_games].to_f).round(2)
+      team[1][:home_scoring_avg] = (team[1][:home_goals] / team[1][:home_games].to_f).round(2)
     end
-    goals_per_game.max_by { |name, goals_per_game| goals_per_game }[0]
+  end
+
+  def best_offense
+    stats_by_team.max_by { |team_id| team_id[1][:total_scoring_avg]}[1][:team_name]
   end
 
   def worst_offense
-    goals_per_game = {}
-    stats_by_team.each do |team|
-      total_goals = (team[1][:home_goals] + team[1][:away_goals]).to_f
-      goals_per_game[team[1][:team_name]] = (total_goals / team[1][:total_games]).round(2)
-    end
-    goals_per_game.min_by { |name, goals_per_game| goals_per_game }[0]
+    stats_by_team.min_by { |team_id| team_id[1][:total_scoring_avg]}[1][:team_name]
+  end
+
+  def best_defense
+    stats_by_team.min_by { |team_id| team_id[1][:total_goals_allowed_avg]}[1][:team_name]
+  end
+
+  def worst_defense
+    stats_by_team.max_by { |team_id| team_id[1][:total_goals_allowed_avg]}[1][:team_name]
+  end
+
```

Comment:
these are pretty clean like these.
starting to wonder if we could do a simple helper method for min and max by because i use those often as well